src/test/java/api/InstancesApiExamples.java

```diff
@@ -873,22 +879,59 @@ public void cannotRemoveHRID() throws Exception {
     assertThat(existingInstance, is(createdInstance));
   }
 
+  @Test
+  public void canFrowardInstanceCreateFailureFromStorage() throws Exception {
+    final String expectedErrorMessage = "Instance-storage is temporary unavailable for create";
+
+    instancesStorageClient.emulateFailure(new EndpointFailureDescriptor()
+      .setFailureExpireDate(DateTime.now(DateTimeZone.UTC).plusSeconds(2).toDate())
+      .setBody(expectedErrorMessage)
+      .setContentType("plain/text")
+      .setStatusCode(500)
+      .setMethod(HttpMethod.POST));
+
+    final Response response = instancesClient.attemptToCreate(smallAngryPlanet(UUID.randomUUID()));
+
+    assertThat(response.getStatusCode(), is(500));
+    assertThat(response.getBody(), is(expectedErrorMessage));
+  }
+
+  @Test
+  public void canFrowardInstanceUpdateFailureFromStorage() throws Exception {
+    final String expectedErrorMessage = "Instance-storage is temporary unavailable for updates";
```

Comment:
How does the system know that the fault is temporary?

Reply:
I'm configuring instance-storage fake API to emulate failures:
```
instancesStorageClient.emulateFailure(new EndpointFailureDescriptor()
      .setFailureExpireDate(DateTime.now(DateTimeZone.UTC).plusSeconds(2).toDate())
      .setBody(expectedErrorMessage)
      .setContentType("plain/text")
      .setStatusCode(500)
      .setMethod(HttpMethod.PUT));
```

This means that each PUT request to /inventory-storage/instances will be rejected with status 500 and with plain/text error saying `Instance-storage is temporary unavailable for updates` till `now() + 2s`.

Reply:
Ok, thanks. This test is checking that the error from storage comes all the way through to the client, is that the case?

Reply:
Yes, correct.